PowerShell-ZAP/PowerShell-ZAP.psm1

```diff
@@ -0,0 +1,377 @@
+# PowerShell-ZAP
+# Copyright (c) 2015 Joona Immonen
+#
+# This is created against 2.4.2 OWASP-ZAP
+# 
+# More information about how to use ZAP API please visit your own zaps local API
+# e.g. http://localhost:8084/UI/ and the documentation
+# https://github.com/zaproxy/zaproxy/wiki/ApiGen_Index
+#
+#Requires -Version 4.0
+
+
+$script:zapBaseAddress = "http://localhost:8084"
+$script:zapLocation = "C:\Program Files (x86)\OWASP\Zed Attack Proxy\"
+$script:zapReportLocation = "C:\Temp\zapresults.xml"
+$script:urlToScan = "http://localhost/"

```

Comment:
Hardcoding stuff like this is bad. First of all, one shouldn't use c:\temp but $env:TEMP instead. Also forcing zapLocation isn't a good idea either: one can install it where ever one wants and in 32-bit Windows there is only Program Files without (x86).

You provide mechanisms to set these below but it would be more reasonable to just stop execution if these aren't set. It would allow a more meaningful error message for the user (eg. "Please set Zap Address before executing with Set-ZapAddress" vs. "zap.exe: The term 'zap.exe' is not recognized...")

I'd like to see these in a configuration file over Set-* functions anyway. Now one has to modify global state to make things to work. This is always more or less evil :)